generator/template/model_template.go

```diff
@@ -207,31 +203,71 @@ func (f TableModelField) TagsString() string {
 
 // Type represents type of the struct field
 type Type struct {
-	ImportPath string
-	Name       string
+	ImportPath            string
+	AdditionalImportPaths []string
+	Name                  string
 }
 
 // NewType creates new type for dummy object
 func NewType(dummyObject interface{}) Type {
+	typeName, importPaths := parseType(dummyObject)
 	return Type{
-		ImportPath: getImportPath(dummyObject),
-		Name:       getTypeName(dummyObject),
+		AdditionalImportPaths: importPaths,
+		Name:                  typeName,
 	}
 }
 
-func getTypeName(t interface{}) string {
-	typeStr := reflect.TypeOf(t).String()
+func parseType(t interface{}) (string, []string) {
+	typ := reflect.TypeOf(t)
+	typeStr := typ.String()
+
+	importPath := typ.PkgPath()
+	if typ.Kind() == reflect.Ptr {
+		importPath = typ.Elem().PkgPath()
+	}
+	importPaths := make(map[string]struct{})
+	if importPath != "" {
+		importPaths[importPath] = struct{}{}
+	}
+
 	typeStr = strings.Replace(typeStr, "[]uint8", "[]byte", -1)
+	firstBracketIdx := strings.Index(typeStr, "[")
+	if firstBracketIdx <= 0 || typeStr[firstBracketIdx-1] == '*' {
```

Comment:
In case of slices first bracket is a at 0, so I assume this will not work for slice of generics - `[]GenericSingle[int]`.